source/blog/2015-11-13-rspec-3-4-has-been-released.md

```diff
@@ -0,0 +1,414 @@
+output. Before RSpec 3.4, this generally worked OK as long as your
+expectation fit on one line. If you formatted it across multiple lines, like
+this:
+
+~~~ ruby
+expect {
+  MyNamespace::MyClass.some_long_method_name(:with, :some, :arguments)
+}.to raise_error(/some error snippet/)
+~~~
+
+...then the failure would just print the first line (`expect {`)
+since that's the stackframe included in the exception. In RSpec 3.4,
+we now load [Ripper](http://ruby-doc.org/stdlib-2.2.0/libdoc/ripper/rdoc/Ripper.html)
+from the standard library if it's available in order to parse the
+source and determine how many lines to include for the full expectation
+expression. For a case like the snippet above, the failure output
+will now include the entire expression.
+
+Thanks to Yuji Nakayama for [implementing](https://github.com/rspec/rspec-core/pull/2083) this!
+
+#### Install `coderay` for Syntax Highlighting
+
+Taking this a step further, if the [`coderay`](http://coderay.rubychan.de/)
+gem is available, RSpec 3.4 will use it to syntax highlight the code snippet
+in your terminal. Here's an example of how that looks, using the code snippet
+from above:
+
+![Failure with syntax highlighting](/images/blog/multiline_failure_with_syntax_highlighting.png)
+
+#### Better Failure Source Detection
+
+RSpec finds the failure code snippet by looking through the exception
+stack trace for an appropriate frame. We could just use the top stack
+frame but that's generally not what you want: when you have an expectation
+failure, the top frame refers to a link in RSpec where the
+`RSpec::Expectations::ExpectationNotMetError` was raised, and you want
+to see the snippet from your `expect` call site rather than seeing
+a snippet of RSpec's code. Before RSpec 3.4, our solution for this
+was fairly naive: we just looked for the first
+stack frame from your spec file. In some situations this would display
+the wrong snippet (such as when your spec file called a helper method
+defined in a `spec/support` file where the real failure occurred).
+In others it didn't find anything and we wound up displaying
+`Unable to find matching line from backtrace` instead.
+
+RSpec 3.4 has much better logic for finding the source snippet: now we
+look for the first frame from the `config.project_source_dirs` (defaults to
+`lib`, `app` and `spec`) and if no matching frame can be found, we fall
+back to the first stack frame. You shouldn't see
+`Unable to find matching line from backtrace` anymore!
+
+### Expectations: Better Compound Failure Messages
+
+Continuing with the "improved failure output" theme, rspec-expectations
+3.4 provides better failure messages for compound expectations. Before,
+we would combine each failure message into a single line. For example,
+this expectation:
+
+~~~ ruby
+expect(lyrics).to start_with("There must be some kind of way out of here")
+              .and include("No reason to get excited")
```

Comment:
It's not clear to me what's the best way to align this.  I aligned `.to` and `.and` but I'm open to an alternate suggestion.  How would you format this?